### PR TITLE
Modify S6684: Improve subscription key detections

### PR DESCRIPTION
**Pull request by loris-s-sonarsource**
## Why
Fixes FP on 
```

              "SonarService:OcpApimSubscriptionKey": "@Microsoft.KeyVault(SecretUri=https://example.vault.azure.net/secrets/Some-Subscription-Key-Dev/e3e56868da48ab516b3538609ec2eb9b)",
              "SonarService:OcpApimSubscriptionKey": "@Microsoft.KeyVault(SecretUri=https://example.vault.azure.net/secrets/Some-Subscription-Key/e3e56868da48ab516b3538609ec2eb9b)",
              "@Microsoft.KeyVault(VaultName=Example;SecretName=Subscription-Key;SecretVersion=e3e56868da48ab516b3538609ec2eb9b)"
```

and generally improves everything

### CODE REVIEW

File: sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/azure.yaml [hunk 1]
```diff
@@ -63,7 +63,7 @@ provider:
         - text: |
             const connStr = "DefaultEndpointsProtocol=https;AccountName=testaccountname;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==";
             const connStr = "DefaultEndpointsProtocol=https;AccountName=testaccountname;AccountKey=C2y6yDjf5/R+ob0N8A7Cgv30VRDJIWEHLM+4QDU5DE2nQ9nDuVTqobD4b8mGGyPMbIZnqyMsEcaGQy67XIw/Jw==";
-          # These are well-known keys used in emulators only
+
```
Comment (loris-s-sonarsource) [inline sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/azure.yaml:66]:
```suggestion
          # These are well-known keys used in emulators only
```

**Review by daniel-teuchert-sonarsource — CHANGES_REQUESTED** [review]
Nice solution!
Just one thing:
Maybe while you are at it add the examples from RSPEC as test cases?

**Comment by loris-s-sonarsource** [comment]
@daniel-teuchert-sonarsource Done, I also made a new PR in rspec:  https://github.com/SonarSource/rspec/pull/3072

**Review by daniel-teuchert-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**